sourcecode/hub/resources/views/content/details.blade.php

```diff
@@ -52,37 +78,52 @@
 {{--    </div>--}}
 
     <x-slot:sidebar>
-        <section>
-            <h2 class="fs-5">{{ trans('messages.version-history') }}
-                <span data-bs-toggle="tooltip" data-bs-placement="top" title="{{ trans('messages.version-history-tooltip')}}">
-                    <x-icon name="info-circle" class="ms-2" />
-                </span>
-            </h2>
-            <ul class="p-0">
-                @foreach ($content->versions->sortByDesc('created_at')->take(3) as $index => $version)
-                    <x-version-details :version="$version" :index="$index" :loop="$loop" />
-                @endforeach
-            </ul>
-
-            @if(count($content->versions) > 3)
-                <button class="btn btn-link d-flex align-items-center" type="button" data-bs-toggle="collapse" data-bs-target="#collapseVersions" aria-expanded="false" aria-label="{{ trans('messages.toggle-listing-of-all-versions')}}">
-                    <x-icon name="chevron-down" class="text-black" aria-hidden="true"/>
-                </button>
-                <div class="collapse" id="collapseVersions">
-                    <ul class="p-0">
-                        @foreach ($content->versions->sortByDesc('created_at')->slice(3) as $index => $version)
-                            <x-version-details :version="$version" :index="$index" :loop="$loop" />
-                        @endforeach
-                    </ul>
-                </div>
-            @endif
-        </section>
+        @can('edit', $content)
+            <section>
+                <h2 class="fs-5">{{ trans('messages.version-history') }}</h2>
+
+                <ul class="list-unstyled d-flex flex-column gap-2 version-history">
+                    @foreach ($content->versions as $v)
+                        @php($isLatest = $content->latestVersion->is($v))
```

Comment:
It's more efficient to put both PHP lines between `@php` and `@endphp`

Reply:
This is technically true, but when we analyse the opcodes generated from the compiled Blade template, we see that the difference is only a single echo statement:

```diff
 EXT_STMT
 ECHO string("                        ")
 EXT_STMT
 T5 = FETCH_OBJ_R CV1($content) string("latestVersion")
 INIT_METHOD_CALL 1 null string("is")
 SEND_VAR_EX CV2($v) 1
 V5 = DO_FCALL
 CV0($isLatest) = QM_ASSIGN V5
 EXT_STMT
-ECHO string("                        ")
-EXT_STMT
 INIT_METHOD_CALL 1 CV4($version) string("is")
 SEND_VAR_EX CV2($v) 1
 V5 = DO_FCALL
 CV3($isCurrent) = QM_ASSIGN V5
 EXT_STMT
```

Nevertheless, it is fixed.